internal/authentication/ldap_client_factory.go

```diff
@@ -117,162 +121,393 @@ type PooledLDAPClientFactory struct {
 	}
 
-	if cap(f.pool) != f.config.Pooling.Count {
-		if err = f.Initialize(); err != nil {
-			return nil, err
+	now := time.Now()
+	deadline := now.Add(f.config.Pooling.Timeout)
+	interval := time.Millisecond * 10
+
+	for time.Now().Before(deadline) {
+		f.wakeupPoolManager() // request more clients if needed
+		select {
+		case <-time.After(interval): // this avoids the blocking time.Sleep()
+			continue // do nothing, except to loop again to signal the pool manager
+		case client = <-pC:
+			if !client.IsHealthy() { // NOTE: we dispose expired clients only on release
+				logging.Logger().Info("Received invalid client from LDAP pool, disposing")
+				_ = f.disposeClient(client)
+				continue
+			}
+			// NOTE: we do NOT check for expiration here - it's a randomized "soft" target.
+			// Rationale: expiration is ony checked on release to avoid potential starvation.
+			// Clients may be used slightly beyond their expiration time, but this is acceptable.
+			logging.Logger().Tracef("Successfully acquired valid client from LDAP pool after %s", time.Since(now))
+			return client, nil
 		}
 	}
+	return nil, fmt.Errorf("timeout waiting for client from LDAP pool")
+}
+
+// must be called EXCLUSIVELY by Initialize() or poolManager()
+func (f *PooledLDAPClientFactory) tryAddPooledClient(pC chan *LDAPClientPooled ) (bool) {
+	attempts := f.config.Pooling.Retries
+	sleep := f.config.Timeout / 8 // initial sleep duration, trade-off between responsiveness and overload
+
+	for !f.isClosing() {
+		start := time.Now()
+		client, err := f.new() // NOTE: new() adheres to config.Timeout, not config.Pooling.Timeout!
+		if err == nil {
+			// Shorten client lifetime by up to 10% to avoid mass exodus events
+			fuzzFactor := 0.90 + 0.1*rand.Float64() // [0.90..1.00)
+			client.expiresAt = time.Now().Add(time.Duration(float64(f.clientLifetime) * fuzzFactor))
+
+			select {
+			case pC <- client:
+				active := f.activeCount.Add(1)
+				logging.Logger().Debugf("Added new client to LDAP pool. Active: %d, Available: %d", active, len(pC))
+				return true
+			default: // shouldn't happen
+				logging.Logger().Warn("LDAP pool closed or full, closing newly created client")
+				_ = client.Close()
+				return false
+			}
+		}
+		logging.Logger().Debugf("Failed to create new client for LDAP pool. Attempt %d/%d, Elapsed: %s: %v",
+			f.config.Pooling.Retries - attempts + 1, f.config.Pooling.Retries + 1, time.Since(start), err)
+
+		if attempts == 0 {
+			logging.Logger().Warningf("Exceeded maximum attempts (%d) for creating pooled LDAP client; last error: %v",
+				f.config.Pooling.Retries + 1, err)
+			return false
+		}
 
-	ctx, cancel := context.WithTimeout(ctx, f.config.Pooling.Timeout)
-	defer cancel()
+		logging.Logger().Tracef("Sleeping %s before next attempt to create pooled LDAP client", sleep)
+		time.Sleep(sleep)
+		sleep *= 2
+		attempts--
+	}
+	return false // pool is closing
+}
 
+// thread-safe, does not block
```

Comment:
Please ensure this starts with a capital and ends with a period, it should also be a sentence or be removed.